plugins/modules/ndo_ipsla_monitoring_policy.py

```diff
@@ -0,0 +1,486 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2024, Shreyas Srish (@shrsr) <ssrish@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: ndo_ipsla_monitoring_policy
+short_description: Manage IPSLA Monitoring Policies on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage IPSLA Monitoring Policies on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Shreyas Srish (@shrsr)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a tenant template.
+    type: str
+    required: true
+  ipsla_monitoring_policy:
+    description:
+    - The name of the IPSLA Monitoring Policy.
+    type: str
+    aliases: [ name ]
+  description:
+    description:
+    - The description of the IPSLA Monitoring Policy.
+    type: str
+  ipsla_monitoring_policy_uuid:
+    description:
+    - The uuid of the IPSLA Monitoring Policy.
+    - This parameter is required when the O(ipsla_monitoring_policy) needs to be updated.
+    type: str
+    aliases: [ uuid ]
+  sla_type:
+    description:
+    - The type of SLA.
+    type: str
+    choices: [ http, tcp, icmp, l2ping ]
+  destination_port:
+    description:
+    - The port used for SLA.
+    - This is only applicable when O(sla_type=tcp).
+    type: int
+  http_version:
+    description:
+    - The HTTP version used for SLA.
+    type: str
+    choices: [ HTTP10, HTTP11 ]
+  http_uri:
+    description:
+    - The URI used for HTTP SLA.
+    type: str
+  sla_frequency:
+    description:
+    - The frequency of SLA monitoring.
+    - The value must be in the range 1 - 300.
+    type: int
+  detect_multiplier:
+    description:
+    - The detection multiplier for SLA.
+    - The value must be in the range 1 - 100.
+    type: int
+  req_data_size:
+    description:
+    - The size of the request data.
+    - The value must be in the range 0 - 17513.
+    type: int
+  type_of_service:
+    description:
+    - The IPv4 Type of Service.
+    - The value must be in the range 0 - 255.
+    type: int
+  operation_timeout:
+    description:
+    - The operation_timeout for SLA.
+    type: int
+  threshold:
+    description:
+    - The threshold for SLA.
+    type: int
```

Comment:
perhaps add the metric? milliseconds?
```suggestion
  operation_timeout:
    description:
    - The operation_timeout for SLA in milliseconds.
    type: int
  threshold:
    description:
    - The threshold for SLA in milliseconds.
    type: int
```